cls/SourceControl/Git/Utils.cls

```diff
@@ -1930,28 +1926,42 @@ ClassMethod SyncIrisWithRepoThroughCommand(ByRef outStream) As %Status
     quit ##class(SourceControl.Git.PullEventHandler).ForModifications(.files)
 }
 
-ClassMethod ExpandClasses(externalName As %String) As %List
+ClassMethod ExpandTruncatedNames(externalName As %String) As %List
 {
     set internalName = $Piece(externalName,".",1,*-1)
     set internalName = $Extract(internalName,4,*)
-    set internalName = $Translate(internalName,"/\%",".."_..PercentClassReplace())
+    set internalName = $Translate(internalName,"/\","..")
+    
+    set externalNameExtension = $Piece(externalName,".",*)
+    if (externalNameExtension = "cls") {
+        set internalName = $Translate(internalName,"%",..PercentClassReplace())
+    }
+    
+    set possibleNames = ""
     do {
-        &sql(select %DLIST(Name) into :classes from %Dictionary.ClassDefinition where Name like '%'||:internalName)
+        if (externalNameExtension = "cls") {
+            &sql(select %DLIST(Name) into :possibleNames from %Dictionary.ClassDefinition where Name like '%'||:internalName)
+        } elseif (externalNameExtension = "lut") {
+            &sql(select %DLIST(distinct TableName) into :possibleNames from Ens_Util.LookupTable where TableName like '%'||:internalName)
```

Comment:
I think some additional work might be needed to generalize this. Two major reasons:
* We want this to work on non-interoperability-enabled namespaces, where Ens_Util.LookupTable isn't available. This is a blocking issue for merge of this PR.
* We want this to work for other document types as well, not just cls and lut. (e.g., I could also see truncation popping up with with .DFI)

There might be an elegant (code-wise) approach using %Library.RoutineMgr:StudioOpenDialog that would hit all relevant cases at once, but I'd be concerned about the possible performance impact. Another approach could be using the [List query of the relevant %Studio.AbstractDocument subclass](https://docs.intersystems.com/irislatest/csp/documatic/%25CSP.Documatic.cls?LIBRARY=%25SYS&CLASSNAME=%25Studio.AbstractDocument#List) directly based on the extension.

Reply:
Thank you for the excellent feedback. Based on this I am thinking that attempting to address this by improving the truncated name expanding logic may not be the ideal approach and I have instead put together #517 to hopefully address this more robustly, at least for pulls initially.